oonib/oonibackend.py

```diff
@@ -102,60 +102,71 @@
     multiService.addService(http_return_request_helper)
     http_return_request_helper.startService()
 
-# add the tor collector service here
+def getHSEndpoint(endpoint_config):
+    hsdir = os.path.join(torconfig.DataDirectory, endpoint_config['hsdir'])
```

Comment:
The `tor_datadir` is actually just the directory where tor is instructed to store it's various files, but it's not really a requirement, nor is it how usually tor is configured, for the hidden service data files to reside in the tor data directory.

I would say we could have a policy of these sorts:

* If `hsdir` is set, then it should be considered as a path that is not relative to `tor_datadir`

* If `hsdir` is not set, then we follow the legacy behaviour of setting the collector `hsdir` to `$tor_datadir/collector` and the bouncer `hsdir` to `$tor_datadir/bouncer`.